### PR TITLE
fix: allow valid css height/width types

### PR DESCRIPTION
**Pull request by nataliepina**
Allow height and width types to accept any valid
CSS height/width properties such as vw, ems,
and other string values beyond numbers.

# Description

I encountered this when attempting to use a value with `vh` where I got a type error. These values should accept any valid related CSS height or width property, such as ems, rems, vh, vw, etc.  Followed the trail up to the `PopoverBox` and update those types there too. 

Included a few other tweaks in here too such as a typo fix, and a className prop for the Dropdown. 

Note: The `React.CSSProperties` type will handle strings or numbers so there will not be a breaking change. 
Emotion will append `px` if only a number is passed in. 

Screen Shot 2022-12-01 at 1 07 09 PM

## Testing

You can update one of these values in the Storybook story. 
I also tested this out using `npm link` in the UI. 

## Checklist

- [ ] This PR is associated with a JIRA and it is mentioned in the commit message footer ("Closes …")
- [ ] Significant changes have been tested downstream to avoid breaking changes
- [ ] This PR contains breaking changes and states in the commit message body ("BREAKING CHANGE: …")
- [x] I have reviewed the changes and provided detail to the sections above

### CODE REVIEW

**Comment by vacas5** [comment]
should it fall back to a string instead of `number`? if someone passes `80` is there code that changes it to `80px`?

**Comment by nataliepina** [comment]
> should it fall back to a string instead of `number`? if someone passes `80` is there code that changes it to `80px`?

I tried to follow the trail of this yesterday and it seems something to do with props being passed between components and how it's handled inside of the Popper and the Dropdownable. I couldn't quite find the exact piece of code setting the px. 
I left number because that does work and it does apply px, that is the previous functionality so I didn't want to cause a breaking change.
However, when I also allow the other valid CSS values as a type they do work as expected, it does not apply any px.
I tested this with `npm link` in the UI and it allowed me to enter both types of values successfully with no type errors.

File: packages/popover/style.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export const menuList = css`
 `;
 
 // Hides unnecessary horizontal scrollbars on Linux and Microsoft
-export const hideHoriztonalScroll = css`
+export const hideHorizontalScroll = css`
```
Comment (clintonmedbery) [inline packages/popover/style.ts:20]: Took me a minute to see the difference 😂

**Review by clintonmedbery — APPROVED** [review]
This is good to me pending Russ' concerns.

**Review by vacas5 — APPROVED** [review]

**Pull request merged**